src/Test/Integration/Block/SSR/RecordListTest.php

```diff
@@ -74,6 +102,8 @@ protected function setUp(): void
         $this->objectManager->addSharedInstance($communicationConfigMock, CommunicationConfig::class);
         $this->objectManager->addSharedInstance($this->redirectMock, Redirect::class);
         $this->objectManager->addSharedInstance($fieldRolesMock, FieldRoles::class);
+
+        $this->recordList = new RecordList($this->createMock(Template\Context::class), $this->createMock(SearchAdapter::class), $this->createMock(SerializerInterface::class), $this->createMock(HttpResponse::class), $this->redirectMock, $fieldRolesMock);
```

Comment:
This is a violation of the integration tests.
We should not manually instantiate the RecordList class - it should be created by Object Manager, as if in real magento application